libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/permission/PermissionsLauncherFragment.kt

```diff
@@ -0,0 +1,65 @@
+package com.mapbox.navigation.dropin.permission
+
+import android.content.Context
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import androidx.activity.result.ActivityResultCallback
+import androidx.activity.result.ActivityResultCaller
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts.RequestMultiplePermissions
+import androidx.core.content.ContextCompat.checkSelfPermission
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+
+/**
+ * A view-less fragment that requests permissions using [ActivityResultCaller] interface.
+ */
+internal class PermissionsLauncherFragment(
+    private val permissions: Array<String>,
+    private val onResult: ActivityResultCallback<Map<String, Boolean>>
+) : Fragment() {
```

Comment:
@tomaszrybakiewicz fragments can't have any arguments in their constructors. This fragment causes 1TAP to crash on every screen rotation:
```
19:33:56.282  E  FATAL EXCEPTION: main
                 Process: com.mapbox.onetap.app, PID: 14241
                 java.lang.RuntimeException: Unable to start activity ComponentInfo{com.mapbox.onetap.app/com.mapbox.onetap.app.core.activity.MainActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment com.mapbox.navigation.dropin.permission.PermissionsLauncherFragment: could not find Fragment constructor
                 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3311)
                 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3460)
                 	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5394)
                 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5302)
                 	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
                 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2047)
                 	at android.os.Handler.dispatchMessage(Handler.java:107)
                 	at android.os.Looper.loop(Looper.java:224)
                 	at android.app.ActivityThread.main(ActivityThread.java:7590)
                 	at java.lang.reflect.Method.invoke(Native Method)
                 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
                 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
                 Caused by: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment com.mapbox.navigation.dropin.permission.PermissionsLauncherFragment: could not find Fragment constructor
                 	at androidx.fragment.app.Fragment.instantiate(Fragment.java:678)
                 	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57)
                 	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:507)
                 	at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:81)
                 	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:85)
                 	at androidx.fragment.app.FragmentManager.restoreSaveStateInternal(FragmentManager.java:2505)
                 	at androidx.fragment.app.FragmentManager.attachController(FragmentManager.java:2665)
                 	at androidx.fragment.app.FragmentController.attachHost(FragmentController.java:117)
                 	at androidx.fragment.app.FragmentActivity.lambda$init$3$androidx-fragment-app-FragmentActivity(FragmentActivity.java:140)
                 	at androidx.fragment.app.FragmentActivity$$ExternalSyntheticLambda3.onContextAvailable(Unknown Source:2)
                 	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.java:99)
                 	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:352)
                 	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:217)
                 	at com.mapbox.onetap.app.core.activity.MainActivity.onCreate(MainActivity.kt:35)
                 	at android.app.Activity.performCreate(Activity.java:7893)
                 	at android.app.Activity.performCreate(Activity.java:7880)
                 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
                 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3286)
                 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3460) 
                 	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5394) 
                 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5302) 
                 	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69) 
                 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
                 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
                 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2047) 
                 	at android.os.Handler.dispatchMessage(Handler.java:107) 
                 	at android.os.Looper.loop(Looper.java:224) 
                 	at android.app.ActivityThread.main(ActivityThread.java:7590) 
                 	at java.lang.reflect.Method.invoke(Native Method) 
                 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539) 
                 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950) 
                 Caused by: java.lang.NoSuchMethodException: com.mapbox.navigation.dropin.permission.PermissionsLauncherFragment.<init> []
                 	at java.lang.Class.getConstructor0(Class.java:2332)
                 	at java.lang.Class.getConstructor(Class.java:1728)
                 	at androidx.fragment.app.Fragment.instantiate(Fragment.java:663)
                 	at androidx.fragment.app.FragmentContainer.instantiate(FragmentContainer.java:57) 
                 	at androidx.fragment.app.FragmentManager$3.instantiate(FragmentManager.java:507) 
                 	at androidx.fragment.app.FragmentState.instantiate(FragmentState.java:81) 
                 	at androidx.fragment.app.FragmentStateManager.<init>(FragmentStateManager.java:85) 
                 	at androidx.fragment.app.FragmentManager.restoreSaveStateInternal(FragmentManager.java:2505) 
                 	at androidx.fragment.app.FragmentManager.attachController(FragmentManager.java:2665) 
                 	at androidx.fragment.app.FragmentController.attachHost(FragmentController.java:117) 
                 	at androidx.fragment.app.FragmentActivity.lambda$init$3$androidx-fragment-app-FragmentActivity(FragmentActivity.java:140) 
                 	at androidx.fragment.app.FragmentActivity$$ExternalSyntheticLambda3.onContextAvailable(Unknown Source:2) 
                 	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.java:99) 
                 	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:352) 
                 	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:217) 
                 	at com.mapbox.onetap.app.core.activity.MainActivity.onCreate(MainActivity.kt:35) 
                 	at android.app.Activity.performCreate(Activity.java:7893) 
                 	at android.app.Activity.performCreate(Activity.java:7880) 
                 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307) 
                 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3286) 
                 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3460) 
                 	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5394) 
                 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5302) 
                 	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69) 
                 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
                 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
                 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2047) 
                 	at android.os.Handler.dispatchMessage(Handler.java:107) 
                 	at android.os.Looper.loop(Looper.java:224) 
                 	at android.app.ActivityThread.main(ActivityThread.java:7590) 
                 	at java.lang.reflect.Method.invoke(Native Method) 
                 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539) 
                 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950) 
19
```

Reply:
Thanks for catching this @Zayankovsky